eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java

```diff
@@ -421,16 +420,27 @@ public void testInvalidConfigJSON() {
     // Create a mock instance of EppoHttpClient
     EppoHttpClient mockHttpClient = mock(EppoHttpClient.class);
 
-    doAnswer(
-            invocation -> {
-              RequestCallback callback = invocation.getArgument(1);
-              callback.onSuccess("{}");
-              return null; // doAnswer doesn't require a return value
-            })
-        .when(mockHttpClient)
-        .get(anyString(), any(RequestCallback.class));
+    when(mockHttpClient.getAsync(anyString()))
+        .thenReturn(CompletableFuture.completedFuture("{}".getBytes()));
+
+    initClient(TEST_HOST, true, true, false, false, mockHttpClient, null, DUMMY_API_KEY, false);
+
+    String result =
+        EppoClient.getInstance()
+            .getStringAssignment("dummy subject", "dummy flag", "not-populated");
+    assertEquals("not-populated", result);
+  }
+
+  @Test
+  public void testInvalidConfigJSONAsync() {
+
+    // Create a mock instance of EppoHttpClient
+    EppoHttpClient mockHttpClient = mock(EppoHttpClient.class);
+    CompletableFuture<byte[]> httpResponse = CompletableFuture.completedFuture("{}".getBytes());
+
+    when(mockHttpClient.getAsync(anyString())).thenReturn(httpResponse);
 
-    initClient(TEST_HOST, true, true, false, false, mockHttpClient, null, DUMMY_API_KEY);
+    initClient(TEST_HOST, true, true, false, false, mockHttpClient, null, DUMMY_API_KEY, false);
 
     String result =
         EppoClient.getInstance()
```

Comment:
This fails with:
```
cloud.eppo.android.EppoClientTest > testInvalidConfigJSON[emulator-5554 - 13] FAILED 
	java.lang.RuntimeException: java.lang.NullPointerException: Attempt to invoke virtual method 'cloud.eppo.ufc.dto.FlagConfig cloud.eppo.api.Configuration.getFlag(java.lang.String)' on a null object reference
	at cloud.eppo.BaseEppoClient.throwIfNotGraceful(BaseEppoClient.java:449)
```

most likely because `BaseEppoClient.getTypedAssignment()` doesn't handle `null` `Configuration`:
```java
  protected EppoValue getTypedAssignment(
      String flagKey,
      String subjectKey,
      Attributes subjectAttributes,
      EppoValue defaultValue,
      VariationType expectedType) {

    throwIfEmptyOrNull(flagKey, "flagKey must not be empty");
    throwIfEmptyOrNull(subjectKey, "subjectKey must not be empty");

    Configuration config = configurationStore.getConfiguration();

    FlagConfig flag = config.getFlag(flagKey); // <- here: calling method on null
```